### PR TITLE
Spot extraction cleaned up

### PR DESCRIPTION
**Pull request by bayliffe**
Replacing PR #86, part of #34.

Testing:
 - [x] Ran tests and they passed OK
 - [x] Added new tests for the new feature(s)

### CODE REVIEW

**Comment by gavinevans** [comment]
@fionaRust #86 has now been approved by two reviewers.

**Comment by Katie-Howard** [comment]
I have looked through all of the code and it seems to make sense to me, and be logical. However I have left a couple of minor comments about some comments in the code. 

I have also run all the tests and it said the unit tests, CLI tests and All tests ran successfully. Codacy and Travis are also fine.

File: lib/improver/tests/spotdata/spotdata/test_common_functions.py [hunk 1]
```diff
@@ -0,0 +1,662 @@
...
+        Case for a time that is available within the additional diagnostic.
+
+        """
+        plugin = extract_ad_at_time
+        result = plugin(self.additional_data, self.time_dt, self.time_extract)
+        self.assertIsInstance(result['air_temperature'], Cube)
+
+    def test_invalid_extraction_time(self):
+        """
+        Case for a time that is available within the additional diagnostic.
```
Comment (bayliffe) [inline lib/improver/tests/spotdata/spotdata/test_common_functions.py]: Yep. Changed.

File: lib/improver/tests/spotdata/spotdata/test_extract_data.py [hunk 2]
```diff
@@ -0,0 +1,600 @@
...
+
+    def test_extracted_value_deep_valley(self):
+        """
+        Test that the plugin returns the correct value.
+
+        Site set to be 100m or 70m below the land surface (90m or 60m below sea
+        level). The enforcement of a maximum extrapolation down into valleys
+        should result in the two site altitudes returning the same temperature.
+
+        This is an extrapolation scenario, an 'unresolved valley'.
```
Comment (bayliffe) [inline lib/improver/tests/spotdata/spotdata/test_extract_data.py]: Yes.

**Pull request merged**